es-vue-base/src/lib-components/EsDataTable.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+    <b-table
+        table-class="font-size-75 font-size-md-100"
+        responsive
+        class="data-table border"
+        :class="{ rounded }"
+        :no-border-collapse="stickyFirstCol"
+        :sticky-header="stickyHeader"
+        :fields="computedFields"
+        :items="items" />
+</template>
+
+<script lang="js">
+export default {
+    name: 'EsDataTable',
+    props: {
+        fields: {
+            type: Array,
+            default: null,
+        },
+        items: {
+            type: Array,
+            required: true,
+        },
+        rounded: {
+            type: Boolean,
+            default: false,
+        },
+        stickyFirstCol: {
+            type: Boolean,
+            default: false,
+        },
+        stickyHeader: {
+            type: [Boolean, String],
+            default: false,
+        },
+    },
+    computed: {
+        computedFields() {
+            // Set 1st column as sticky if property set
+            if (!this.fields) {
+                // Leave fields to be interpolated if not provided
+                return null;
+            }
+            return this.fields.map((field, index) => {
+                if (index === 0) {
+                    let fieldObj;
+                    if (typeof field === 'string') {
+                        fieldObj = { key: field };
+                    } else {
+                        fieldObj = { ...field };
+                    }
+                    fieldObj.stickyColumn = this.stickyFirstCol;
+                    return fieldObj;
+                }
+                return field;
+            });
+        },
+    },
+};
+</script>
+<style lang="scss" scoped>
+@import '~@energysage/es-bs-base/scss/bootstrap';
+@import '~@energysage/es-bs-base/scss/variables';
+
+.data-table {
```

Comment:
This SCSS could use some optimization. 

The `th` and `td` variables have the same definition aside from `font-weight`. Combining them and setting `font-weigh: bold` for `th` only would make this easier to maintain/grep.

We should be able limit the number of `::v-deep` selectors we need;

```css
.data-table {
  ::v-deep {
     th, td {
```